### PR TITLE
MOTR secret management

### PR DESCRIPTION
**Pull request by piyushm82**

### CODE REVIEW

File: iam_policies/kms_key_policy.json.tpl [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "${kms_role_arn}"
+      },
+      "Action": [ 
+        "kms:Enable*",
+        "kms:Get*",
+        "kms:Describe*",
+        "kms:Encrypt",
+        "kms:Decrypt",
```
Comment (przemos) [inline iam_policies/kms_key_policy.json.tpl]: kms:Decrypt should be the only permission we need here

File: iam_policies/kms_key_policy.json.tpl [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        "kms:ReEncrypt*"
+      ],
+      "Resource": "*"
+    },
+    {
+      "Sid": "Enable key management Permissions",
```
Comment (przemos) [inline iam_policies/kms_key_policy.json.tpl]: Root does not have it anyway ?

File: variables.tf [hunk 3]
```diff
@@ -252,3 +252,15 @@ variable "base_url" {
   type        = "string"
   description = "MOT reminder base url"
 }
+
+variable "kms_key_rotation" {
+  type        = "string"
+  description = "KMS key rotation enable"
+  default     = "false"
```
Comment (sebolabs) [inline variables.tf]: no defaults in the module, only on the component side

File: 03_secrets.tf [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+resource "aws_kms_key" "MOTR_Lambda_Key" {
+    description              = "MOTR_Key-${var.environment}"
+    enable_key_rotation      = "${var.kms_key_rotation ? true : false}"
+    deletion_window_in_days  = "${var.kms_deletion_window}"
+    policy                   = "${data.template_file.kms_lambda_policy.rendered}"
+}
+
+resource "aws_kms_alias" "MOTR_Lambda_Alias" {
+  name = "alias/MOTR-${var.environment}"
```
Comment (sebolabs) [inline 03_secrets.tf]: no capital letters in the name attrib, just to be consistent across all our code

File: 03_secrets.tf [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  template = "${file("${path.module}/iam_policies/kms_key_policy.json.tpl")}"
+
+  vars {
+    kms_role_arn = "${aws_iam_role.MotrWebAppLambda.arn}"
+    account_id   = "${data.aws_caller_identity.current.account_id}"
+  }
+}
+
+resource "aws_kms_key" "MOTR_Lambda_Key" {
+    description              = "MOTR_Key-${var.environment}"
```
Comment (sebolabs) [inline 03_secrets.tf]: no need for _ and - in the description - just to be consistent across our code and your code :P

File: 03_secrets.tf [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    enable_key_rotation      = "${var.kms_key_rotation ? true : false}"
+    deletion_window_in_days  = "${var.kms_deletion_window}"
+    policy                   = "${data.template_file.kms_lambda_policy.rendered}"
+}
```
Comment (sebolabs) [inline 03_secrets.tf]: indent 2 spaces not 4 :)

**Pull request merged**